lib/repl.js

```diff
@@ -386,14 +386,15 @@ function REPLServer(prompt,
   self.bufferedCommand = '';
   self.lines.level = [];
 
-  function complete(text, callback) {
-    self.complete(text, callback);
-  }
+  // Figure out which "complete" function to use.
+  self.completer = (typeof options.completer === 'function')
+    ? options.completer
+    : complete.bind(self);
```

Comment:
@cjihrig You're right. I delve a little more and now I know exactly why it works without the extra binding:

1- In the following code snippet `complete` will set the correct `this` to `self` since it is defined as part of the object `self`, so it doesn't matter the implementation, but the object that defined it as part of one of its attributes.

That will allow to call `REPL.completer` always with the correct binding.

```js
self.completer = (typeof options.completer === 'function')
    ? options.completer
    : complete;
```

2- The next snippet, enforces the `this` on `Interface`, so it does resets the internal `this` used by `Interface.completer` in the `readline` module, so it will always work: so no extra binding needed.

```js
Interface.call(this, {
    input: self.inputStream,
    output: self.outputStream,
    completer: self.completer,
    terminal: options.terminal,
    historySize: options.historySize,
    prompt
  });
```

3- As long as the `complete` function is not called by directly, but through the `REPL` related method, the `this` will have the correct reference, and since all tests use `replInstance.complete`, then it works.

So, in conclusion, it seems that while 3 is enforced, meaning no one calls directly `complete` without setting the proper this, inside the repl file, it is not needed the extra binding :) so I will remove it and update the thread.